### PR TITLE
Add support for creating system-probe configuration

### PR DESCRIPTION
**Pull request by kevinconaway**
This PR adds support for writing the `system-probe.yaml` configuration file to the ansible playbook.

If the system probe is enabled, the corresponding service will be started.

@DataDog/burrito 
@hush-hush

### CODE REVIEW

File: tasks/agent6-linux.yml
Comment (leeavital) [inline tasks/agent6-linux.yml]: should this be owned by dd-agent ? The process-agent needs to read the system-probe.yaml file, so making the owner root could be a problem

Comment (kevinconaway) [reply tasks/agent6-linux.yml]: Its currently owned by root in our environments.

Comment (sunhay) [reply tasks/agent6-linux.yml]: I do think it should be owned by root - is there a way to confirm that it'll be readable by dd-agent? (or additional config to change the chmod settings?)

File: README.md
Comment (hush-hush) [inline README.md]: Please move this section after the `datadog_config` one. We try to have the most used option first.
Moving it line 274 or 279 would be better.

File: README.md
Comment (hush-hush) [inline README.md]: Please update the TOC of the readme too

File: README.md
Comment (hush-hush) [inline README.md]: Could you add a link to the official system-probe doc page. Also please mention that this is only supported for Agent6 Linux.

File: tasks/agent6-linux.yml
Comment (hush-hush) [inline tasks/agent6-linux.yml]: Is the system prob only available one linux ? Do we plan to have support for Windows at some point ?

Please move this after line 52 (before or after `Create trace agent configuration file`). We want to create the datadog.yaml first (in case it fails the error raise to the customer will be about the main agent).

File: tasks/agent6-linux.yml
Comment (hush-hush) [reply tasks/agent6-linux.yml]: I was told it will be own by root as a security requirement. If it's own by dd-agent then we must move it inside the datadog.yaml like every other "datadog" configuration.

Comment (hush-hush) [inline tasks/agent6-linux.yml]: Is there a situation where a user might want to change this ? Should we make it configurable ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: tasks/agent6-linux.yml
Comment (kevinconaway) [reply tasks/agent6-linux.yml]: > Is the system prob only available one linux ? 

Yes

> Do we plan to have support for Windows at some point ?

I believe so.

File: tasks/agent6-linux.yml
Comment (kevinconaway) [reply tasks/agent6-linux.yml]: Not to my knowledge but I suppose if someone is using linux capabilities, they may create a different user for this purpose.  It can't hurt to template it out.

**Comment by kevinconaway** [comment]
@sunhay Per our convo, I've updated this PR to also start the probe if it is enabled.

File: tasks/agent6-linux.yml
Comment (hush-hush) [reply tasks/agent6-linux.yml]: Well, templating it means we support if officially, I let you choose if it's something we want to do.

File: tasks/agent6-linux.yml
Comment (hush-hush) [inline tasks/agent6-linux.yml]: why are we checking twice that the system-probl is is running ? Since it's added to the list line 93

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: tasks/agent6-linux.yml
Comment (sunhay) [inline tasks/agent6-linux.yml]: Will this still work on older agent versions that don't have the system probe bundled?

File: tasks/agent6-linux.yml
Comment (kevinconaway) [reply tasks/agent6-linux.yml]: @sunhay Do we want this templated?

File: tasks/agent6-linux.yml
Comment (hush-hush) [reply tasks/agent6-linux.yml]: Indeed, I don't think it will

Comment (kevinconaway) [reply tasks/agent6-linux.yml]: Good catch.  I've updated this to use the [service_facts](https://docs.ansible.com/ansible/latest/modules/service_facts_module.html#service-facts-module) module to determine if the service is present.

File: tasks/agent6-linux.yml
Comment (kevinconaway) [reply tasks/agent6-linux.yml]: Removed

**Review by sunhay — APPROVED** [review]
LGTM

We should wait on @hush-hush's review as well before merging :)

**Comment by kevinconaway** [comment]
Per our convo yesterday, I am going to update the perms to be `0640` and the group owner to be the agent user.  I am testing out internally first before doing so.

**Review by hush-hush — APPROVED** [review]

File: tasks/agent6-linux.yml
Comment (stefanux) [reply tasks/agent6-linux.yml]: > I was told it will be own by root as a security requirement. If it's own by dd-agent then we must move it inside the datadog.yaml like every other "datadog" configuration.

what is the security-gain of a root-ownership especially on this file? if your thinking is: "the unpriviledged datadog-user should not be able to modify its own configuration" than *all* config should be owned by root and only readable by the datadog_group.

Comment (sunhay) [reply tasks/agent6-linux.yml]: The system-probe needs to run with elevated privileges as it uses [eBPF](https://lwn.net/Articles/740157/). 

We do not want the unprivileged datadog-user to make changes to a configuration file that a privileged binary reads - though since the unprivileged user still needs to access it, it should have read-only rights.

**Pull request merged**
